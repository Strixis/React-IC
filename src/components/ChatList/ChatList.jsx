import './ChatList.less';

import React, { Component } from 'react';
import PropType from 'prop-types';

import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

class ChatList extends Component {
  static propTypes = {
    newChat: PropType.func,
    chats: PropType.array,
  }

  render() {
    // const { chats, newChat, navigate, removeChat } = this.props;

    // return (
    //   <div className="chat-list">
    //     <List className="list">
    //       { chats.map((chat) => {
    //         return (
    //           <div key={ chat.id }>
    //             <button onClick={ () => navigate(chat.id) } className="list_link" >
    //               <ListItem style={{ textAlign: "center" }}>
    //                 <ListItemText primary={ chat.name } />
    //               </ListItem>
    //             </button>
    //             <button onClick={ () => removeChat(chat.id) }>remove</button>
    //           </div>
    //         )
    //       })}
    //     </List>
    //     <button onClick={ newChat }>New chat</button>
    //   </div>
    // )
    const { chats, newChat, navigate } = this.props;

    return (
      <div className="chat-list">
        <List className="list">
          { chats.map((chat) => {
            return (
              <div key={ chat.id }>
                <button onClick={ () => navigate(chat.id) } className="list_link" >
                  <ListItem style={{ textAlign: "center" }}>
                    <ListItemText primary={ chat.name } />
                  </ListItem>
                </button>
              </div>
            )
          })}
        </List>
        <button onClick={ newChat }>New chat</button>
      </div>
    )
  }
}

export { ChatList };
