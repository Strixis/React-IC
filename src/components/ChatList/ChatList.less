.chat-list {
  width: 20%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0 20px;
  box-sizing: border-box;
}

.list {
  width: 100%;
  height: 100%;
  overflow-y: auto;
  &_link {
    display: block;
    background-color: aliceblue;
    border-radius: 30px;
    margin: 10px 0;
  }
}
