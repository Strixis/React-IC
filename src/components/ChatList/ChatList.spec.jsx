import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { ChatList } from './ChatList';

test('ChatList should be required', () => {
  const tree = renderer.create(<ChatList chats={[
    {name: "Chat 1", id: "1"},
    {name: "Chat 2", id: "2"},
    {name: "chat 3", id: "3"}
  ]} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('ChatList should be rendered with 3 .list_link element', () => {
  const chatList = shallow(<ChatList chats={[
    {name: "Chat 1", id: "1"},
    {name: "Chat 2", id: "2"},
    {name: "chat 3", id: "3"}
  ]} />);
  const lis = header.find('.list_link');

  expect(lis.length).toBe(3);
});
