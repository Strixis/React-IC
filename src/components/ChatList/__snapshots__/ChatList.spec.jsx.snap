// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be required 1`] = `
<div
  className="chat-list"
>
  <ul
    className="MuiList-root list MuiList-padding"
  >
    <div>
      <button
        className="list_link "
        onClick={[Function]}
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Chat 1
            </span>
          </div>
        </li>
      </button>
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div>
      <button
        className="list_link "
        onClick={[Function]}
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Chat 2
            </span>
          </div>
        </li>
      </button>
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div>
      <button
        className="list_link "
        onClick={[Function]}
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              chat 3
            </span>
          </div>
        </li>
      </button>
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
  </ul>
  <button>
    New chat
  </button>
</div>
`;
